sdk/core/core-client-rest/test/sendRequest.spec.ts

```diff
@@ -83,6 +83,29 @@ describe("sendRequest", () => {
         contentType: "multipart/form-data",
       });
     });
+
+    it("should handle request body as FormData with array of binary", async () => {
+      const expectedFormData = { fileName: "foo.txt" };
+      const mockPipeline: Pipeline = createEmptyPipeline();
+      mockPipeline.sendRequest = async (_client, request) => {
+        assert.equal(request.formData?.fileName, "foo.txt");
+        assert.isArray(request.formData?.files);
+        const files = request.formData?.files as Blob[];
+        assert.lengthOf(files, 2);
+
+        assert.instanceOf(files[0], Blob);
+        assert.instanceOf(files[1], Blob);
+        assert.sameOrderedMembers([...new Uint8Array(await files[0].arrayBuffer())], [...foo]);
+        assert.sameOrderedMembers([...new Uint8Array(await files[1].arrayBuffer())], [...foo]);
+
+        return { headers: createHttpHeaders() } as PipelineResponse;
+      };
+
+      await sendRequest("POST", mockBaseUrl, mockPipeline, {
+        body: { ...expectedFormData, files: [foo, foo] },
+        contentType: "multipart/form-data",
```

Comment:
One thing I think we are lack of testing to verify the `raw` content of request.

We may not need that kind of `raw` binary but we need to ensure the relevant content and their organization are as our expectation.

Reply:
The building of the request body on the wire is the responsibility of `core-rest-pipeline`'s `formDataPolicy` and `multipartPolicy`. We have test coverage of that there.

Reply:
Logged an issue to improve coverage of the core-rest-pipeline formDataPolicy/multipartPolicy: https://github.com/Azure/azure-sdk-for-js/issues/28355